### PR TITLE
Feat zs 183

### PR DESCRIPTION
**Pull request by bjornamkreutz**
ZS-183

### CODE REVIEW

**Comment by bjornamkreutz** [comment]
Added @skip to ZOWN test: setData for Boiler,HP and Battery. When merged Protocol-adapter-IEC61850 to development, @skip will be removed.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    private boolean isInitialised() {
+        return this.rtuSimulator != null;
+    }
+
+    public Iec61850MockServerMarkerWadden() {
+    }
+
+    public Iec61850MockServerMarkerWadden(final String serverName, final String icdFilename, final int port) {
+        super();
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java]: Is this super call needed here?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/mocks/RtuSimulatorSteps.java [hunk 2]
```diff
@@ -77,7 +81,7 @@ public void theRtuSimulatorShouldContain(final List<List<String>> mockValues) th
             final String logicalDeviceName = mockValue.get(INDEX_LOGICAL_DEVICE_NAME);
             final String node = mockValue.get(INDEX_NODE_NAME);
             final String value = mockValue.get(INDEX_NODE_VALUE);
-            this.mockServer.assertValue(logicalDeviceName, node, value);
+            this.mockServerMarkerWadden.assertValue(logicalDeviceName, node, value);
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/mocks/RtuSimulatorSteps.java]: Restore original code this.mockServer
Create a new test step "a MarkerWadden RTU simulator returning" for the mockServerMarkerWadden (maybe even a new MarkerWaddenSimulatorSteps class?)

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+     * supplied parameters in the feature file. Only the provided feature
+     * parameters will override the default parameter from the property file.
+     * Currently the properties can be overwritten: ServerName, Port and/or
+     * IcdFilename.
+     *
+     * @param settings
+     *            the feature file parameters.
+     *
+     */
+    public void restart(final Map<String, String> settings) {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java]: Remove this method, as it was not correctly working anyway and no longer needed

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature [hunk 4]
```diff
@@ -1,16 +1,17 @@
-@Iec61850MockServer
+@Skip
+@Iec61850MockServerMarkerWadden
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature]: Create 2 separate feature files, 1 for the Pampus RTU and 1 for the MarkerWadden RTU?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import org.springframework.stereotype.Component;
+
+import com.alliander.osgp.simulator.protocol.iec61850.server.RtuSimulator;
+
+@Component
+public class Iec61850MockServerMarkerWadden {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Iec61850MockServerMarkerWadden.class);
+
+    @Value("${iec61850.mock.icd.filename.marker.wadden:MarkerWadden_0_1_1_reporting_hack.icd}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java]: Be consistent, either use MarkerWadden_0_1_1_reporting_hack.icd or MarkerWadden_0_1_1.icd

File: cucumber-tests-platform/src/test/resources/cucumber-platform.properties [hunk 6]
```diff
@@ -61,6 +61,8 @@ response.delay.random.range=2000
 #IEC61850
 #ICD file from osgp-protocol-simulator-iec61850 resources
 iec61850.mock.icd.filename=Pampus_v0.4.5_reporting_hack.icd
+iec61850.mock.icd.filename.marker.wadden=MarkerWadden_0_1_1.icd
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/cucumber-platform.properties]: Be consistent, either use MarkerWadden_0_1_1_reporting_hack.icd or MarkerWadden_0_1_1.icd

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java (hunk 5 above)
Comment (bjornamkreutz) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/mocks/iec61850/Iec61850MockServerMarkerWadden.java]: Uses now MarkerWadden_0_1_1.icd

File: cucumber-tests-platform/src/test/resources/cucumber-platform.properties (hunk 6 above)
Comment (bjornamkreutz) [inline cucumber-tests-platform/src/test/resources/cucumber-platform.properties]: Uses now MarkerWadden_0_1_1.icd

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature [hunk 7]
```diff
@@ -4,26 +4,26 @@ Feature: MicroGrids Set Data Service
...
-      | DeviceIdentification | RTU10003                              |
+      | DeviceIdentification       | RTU10003            |
     When a set data request is received    
       | DeviceIdentification       | RTU10003            |
       | NumberOfSystems            |                   1 |
       | SystemId_1                 |                   1 |
       | SystemType_1               | <SystemType_1>      |
       | NumberOfSetPoints_1        |                   3 |
-      | SetPointId_1_1             |                   1 |
+      | SetPointId_1_1             | <SetPointId_1_1>    |
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature]: Refactor feature file to use only 1 NodeId_1 to replace SetPointId and ProfileId

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature [hunk 8]
```diff
@@ -41,22 +41,36 @@ Feature: MicroGrids Set Data Service
...
-      | <Logical_node> | DSCH1.SchdAbsTm.val.0  |                  74 |
-      | <Logical_node> | DSCH1.SchdAbsTm.time.0 | 2016-11-21T01:45:00 |
-      | <Logical_node> | DSCH1.SchdAbsTm.val.1  |                  45 |
-      | <Logical_node> | DSCH1.SchdAbsTm.time.1 | 2016-11-21T12:45:00 |
-      | <Logical_node> | DSCH1.SchdAbsTm.val.2  |                  15 |
-      | <Logical_node> | DSCH1.SchdAbsTm.time.2 | 2016-11-22T00:00:00 |
-      | <Logical_node> | DSCH1.SchdAbsTm.val.3  |                  21 |
-      | <Logical_node> | DSCH1.SchdAbsTm.time.3 | 2016-11-22T01:30:00 |
+    And a MarkerWadden RTU simulator should be returning
+      | <Logical_device> | <DSCH>.SchdId.setVal    |                   1 |
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature]: Refactor `<DSCH>` to `DSCH<NodeId_1>`

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/mocks/RtuSimulatorSteps.java [hunk 9]
```diff
@@ -85,7 +85,7 @@ public void theRtuSimulatorShouldContain(final List<List<String>> mockValues) th
         }
     }
 
-    @Then("^a MarkerWadden RTU simulator should be returning$")
+    @Then("^a MarkerWadden RTU simulator should contain$")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/mocks/RtuSimulatorSteps.java]: **the Marker Wadden** RTU simulator should contain

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature [hunk 10]
```diff
@@ -41,36 +41,36 @@ Feature: MicroGrids Set Data Service
...
-      | <Logical_device> | <DSCH>.SchCat.setVal    |                   1 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.val.0  |                  74 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.time.0 | 2016-11-21T01:45:00 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.val.1  |                  45 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.time.1 | 2016-11-21T12:45:00 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.val.2  |                  15 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.time.2 | 2016-11-22T00:00:00 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.val.3  |                  21 |
-      | <Logical_device> | <DSCH>.SchdAbsTm.time.3 | 2016-11-22T01:30:00 |
+    And a MarkerWadden RTU simulator should contain
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/SetData.feature]: **the Marker Wadden** RTU simulator should contain

**Comment by smvdheijden** [comment]
retest this please

**Comment by bjornamkreutz** [comment]
retest this please

**Comment by smvdheijden** [comment]
Test failures are unrelated to this feature

**Pull request merged**